File: c++/1221. Split a String in Balanced Strings/1221.cpp
class Solution {
public:
    int balancedStringSplit(string s) { 
        int balanced=0,count=0; //define 2 new ints
        for (int i=0;i<s.size();i++){ //for looping into s.size
            if (s[i] == 'R') //count Rs and Ls
                balanced+=1; //if its R we goin +1 to balanced
            if (s[i] == 'L') 
                balanced-=1; //else -1
            if (balanced == 0)
                count+=1; //and if we get balanced if 0 we should add 1 to our result counter
        }
        return count;
    }
};

File: python/1221. Split a String in Balanced Strings/1221.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        count = {'R' : 0, 'L' : 0} #we will be count how much Ls and Rs in string
        res = 0 #if count of Rs and Ls are equal we need to +1 to res
        for i in s:
            if i == 'R':
                count['R'] += 1
            elif i == 'L':
                count['L'] += 1
            if count['R'] == count['L']:
                res += 1
        return res
